### PR TITLE
feat: enable destination config override

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

Complete config override implementation in DeviceModeDestinations plugin, enabling runtime destination configuration changes via sourceConfigurationOverride.destinations[].config.

- Enhanced applyOverrideToDestination() to support config property overrides with deep cloning
- Added property removal when override values are null/undefined
- Added comprehensive config override test coverage
- Destinations now marked as overridden: true when modified

I've also enriched the memory bank with additional content.

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-3326/add-support-to-dynamically-override-destination-configuration

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **Documentation**
  - Added and extensively expanded documentation in the Memory Bank, including new guides on code structure, technical stack, system architecture, Git workflow, CI/CD, task implementation workflow, and maintenance instructions.
  - Restructured the Memory Bank README for improved navigation and onboarding.
- **Bug Fixes**
  - Improved configuration override logic for device mode destinations to ensure overrides are applied correctly only when destinations are enabled.
- **Tests**
  - Enhanced test coverage for configuration overrides, including edge cases for property removal, type changes, and enabled status handling.
- **Chores**
  - Increased the size limit for the "Core (Content Script) - Modern - NPM (CJS)" bundle to 40 KiB.

### CODE REVIEW

File: packages/analytics-js-plugins/__tests__/deviceModeDestinations/utils.test.ts [hunk 1]
```diff
@@ -594,5 +593,89 @@ describe('deviceModeDestinations utils', () => {
...
+      };
+      const result = applyOverrideToDestination(mockDestination, override);
+
+      expect(result).not.toBe(mockDestination);
+      expect(result.overridden).toBe(true);
+      expect(result.config.apiKey).toEqual({ nested: 'object' });
+      expect(result.config.blacklistedEvents).toBe('stringValue');
+    });
+
+    it('should combine enabled and config overrides', () => {
```
Comment (MoumitaM) [inline packages/analytics-js-plugins/__tests__/deviceModeDestinations/utils.test.ts]: If the enabled flag is false, there is no benefit in updating the config.

Comment (saikumarrs) [reply packages/analytics-js-plugins/__tests__/deviceModeDestinations/utils.test.ts]: Not critical but I've addressed it anyway.

**Review by MoumitaM — APPROVED** [review]
